plugins/fused-ops-and-kernels/src/fms_acceleration_foak/framework_plugin_fast_quantized_peft.py

```diff
@@ -54,11 +54,31 @@ def _all_reduce_hook(grad):
 
         # because we will ignore these from FSDP, we need to manually
         # move them to gpu if they are already not on them
+        # - if the adapters are on meta, we assume that this is for FSDP
+        #   low_cpu_mem_mode purposes, and that the values will be synced over
+        # - So just initialize them to empty.
         if not A.weight.is_cuda:
-            set_module_tensor_to_device(A, "weight", "cuda")
+            value = A.weight
+
+            if is_fsdp_enabled() and value.device == torch.device('meta'):
+                # if low_cpu_mem_mode
+                value = torch.empty(*value.size(), dtype=value.dtype)
```

Comment:
this initializes the weight in memory, but sets to to all zeros. This is done for `rank > 0`, because in `low_cpu_mem_usage=True`, all params are initialized in the `meta` device. This step is needed to realize the param.